api/controllers/pinsV2.js

```diff
@@ -0,0 +1,301 @@
+// Imports
+const defaultLog = require('winston').loggers.get('default');
+const Actions    = require('../helpers/actions');
+const projectDAO = require('../dao/projectDAO');
+const pinDAO     = require('../dao/pinDAO');
+
+// Constants
+const PUBLIC_ROLES = ['public'];
+const SECURE_ROLES = ['sysadmin', 'staff'];
+// Vars
+/* put any needed local variables here */
+// functions
+/* put any needed local functions here */
+// Exports
+
+// OPTIONS
+exports.pinOptions = function (args, res, rest) 
+{
+  res.status(200).send();
+};
+
+exports.pinOptionsProtected = function (args, res, rest) 
+{
+  res.status(200).send();
+};
+
+// GET (Public, getPin)
+exports.fetchPins = async function (args, res, next)
+{
+    defaultLog.debug('>>> {GET}/Public/Projects/{id}/Pins');
+
+    try
+    {
+        if (args.swagger.params.hasOwnProperty('projId'))
+        {
+            let projectId = args.swagger.params.projId.value;
+                
+            defaultLog.debug(' Fetching pins for project: ' + projectId);
+            
+            let project = await projectDAO.getProject(SECURE_ROLES, projectId);
+
+            if(project)
+            {
+                let pageNumber = args.swagger.params.hasOwnProperty('pageNumber') && args.swagger.params.pageNumber.value ? args.swagger.params.pageNumber.value : 1;
+                let pageSize   = args.swagger.params.hasOwnProperty('pageSize')   && args.swagger.params.pageSize.value   ? args.swagger.params.pageSize.value   : 10;
+                let sortBy     = args.swagger.params.hasOwnProperty('sortBy')     && args.swagger.params.sortBy.value     ? args.swagger.params.sortBy.value     : '';
+    
+                let data = await pinDAO.getProjectPins(PUBLIC_ROLES, project, pageNumber, pageSize, sortBy);
+    
+                return data ? Actions.sendResponseV2(res, 200, data) 
+                            : Actions.sendResponseV2(res, 404, { code: 404, message: 'Project pins were not found'});
+            }
+            else
+            {
+                return Actions.sendResponseV2(res, 404, { status: 404, message: 'Project ' + projectId + ' not found.'});
+            }
+        }
+        else
+        {
+            throw Error('Invalid request');
+        }
+    }
+    catch (e)
+    {
+        defaultLog.error('### Error in {GET}/Public/Projects/{id}/Pins :', e);
+        return Actions.sendResponseV2(res, 500, { code: '500', message: 'Internal Server Error', self: 'Api/Public/Projects/{id}/Pins' });        
+    }
+    finally
+    {
+        defaultLog.debug('<<< {GET}/Public/Projects/{id}/Pins');
```

Comment:
This debug message is already getting recorded at the top of the function.

Reply:
Actually, it's an "exit" debug message. note the "<<<" and ">>>" in the comments. I can remove them if it's preferred; they're just debug log helpers to let you know when a method started and when a method ended.

Reply:
Ahh, that make sense.